snapcraft/storeapi/__init__.py

```diff
@@ -432,6 +449,42 @@ def snap_release(self, snap_name, revision, channels):
 
         return response_json
 
+    def generate_snap_build(self, authority_id, snap_id, snap_name,
+                            snap_filename, grade, key_name, assertion):
+        with open(assertion, 'w+') as outfile:
+            try:
+                # snap-digest is calculated by snap sign-build itself
+                # snap-size is calculated by snap sign-build itself
+                # timestamp is taken by snap sign-build itself
+                cmd = ['snap', 'sign-build',
```

Comment:
You'll need to update a few package relationships to refer to `snapd (>= 2.15~)` rather than `snapd (>= 2.14.2~)` in order to use this: at least the one in `debian/control` and the first one in `debian/tests/control`, and possibly the second in that latter file as well.